code/modules/clothing/masks/gasmask.dm

```diff
@@ -26,10 +26,19 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
 	var/starting_filter_type = /obj/item/gas_filter
 	///Does the mask have an FOV?
 	var/has_fov = TRUE
+	///Cigarette in the mask
+	var/obj/item/clothing/mask/cigarette/cig
 
 /datum/armor/mask_gas
 	bio = 100
 
+/obj/item/clothing/mask/gas/worn_overlays(mutable_appearance/standing, isinhands)
+	. = ..()
+	if(!isinhands && cig)
+		. += cig.build_worn_icon(default_layer = FACEMASK_LAYER, default_icon_file = 'icons/mob/clothing/mask.dmi')
+	else
+		cut_overlays()
```

Comment:
Do you know if this `cut_overlays` is necessary? Seems like it'd end up killing any other overlays added to the mask unnecessarily (since the cig overlay is simply not added if there is no cig)